core/logger/logger.go

```diff
@@ -208,14 +208,17 @@ func (l *zapLogger) Sync() error {
 }
 
 // newProductionConfig returns a new production zap.Config.
-func newProductionConfig(dir string, jsonConsole bool, toDisk bool, unixTS bool) zap.Config {
+func newProductionConfig(dir string, jsonConsole bool, toDisk bool, unixTS bool, toSplunk bool) zap.Config {
 	config := newBaseConfig()
 	if !unixTS {
 		config.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
 	}
 	if !jsonConsole {
 		config.OutputPaths = []string{"pretty://console"}
 	}
+	if toSplunk {
+		config.OutputPaths = append(config.OutputPaths, "splunk://hec")
```

Comment:
I am not too familiar with Splunk, but can this url change? Perhaps we want to specify `LOG_SPLUNK_URL` instead and have it be configurable?

Reply:
This is an internal zap convention. The `splunk` scheme is registered, similar to what you do for `pretty` console above. The zap logger then generates loggers on demand.